extensions/widgets/segment_new/segment_new.lcb

```diff
@@ -0,0 +1,716 @@
+			end if
+			--
+			
+			-- draw the label(s) in the selected segment(s)
+			if mSegmentDisplay is "icon" then
+				execute script ("return numtocodepoint(" & element tX of mSelectedIcons & ")")
+				put the result into tLabel
+			else
+				put (element tX of mSegmentLabels) into tLabel
+			end if
+			
+			set the paint of this canvas to fetchPaint("selected label")
+			fill text tLabel at center of fetchLabelRect(tWidth) on this canvas
+			--
+		else
+			
+			-- draw the label(s) in the non-selected segment(s)
+			if mSegmentDisplay is "icon" then
+				execute script ("return numtocodepoint(" & element tX of mSegmentIcons & ")")
+				put the result into tLabel
+			else
+				put (element tX of mSegmentLabels) into tLabel
+			end if
+			
+			set the paint of this canvas to fetchPaint("label")	
+			fill text tLabel at center of fetchLabelRect(tWidth) on this canvas
+			--
+			
+		end if
+		add tWidth to mLeft
+	end repeat
+	
+end handler
+----------
+
+----------
+-- this handler updates the mLines variable, a list of line paths that separate the segments
+private handler updateLines() as undefined
+
+	variable tX as integer
+	variable tWidth as real
+	variable tLeft as real
+	
+	put 0 into tLeft
+	
+	repeat with tX from 1 up to mNumSegments
+	
+		put fetchWidth(tX) into tWidth
+		add tWidth to tLeft
+		if tX < mNumSegments then
+			push line path from point [tLeft, 0] to point [tLeft, my height] onto back of mLines
+		end if
+		
+	end repeat
+end handler
+----------
+
+----------
+-- this handler updates the mPerimeter variable, a rectangle that is the perimeter of the control
+private handler updatePerimeter() as undefined
+	variable tRight as number
+	variable tX as integer
+	
+	put 0 into tRight
+	repeat with tX from 1 up to mNumSegments
+	 	add fetchWidth(tX) to tRight
+	end repeat
+	
+	put rounded rectangle path of rectangle [0.5, 0.5, (the trunc of tRight) + 0.5, (the trunc of my height) - 0.5] with radius mRadius into mPerimeter
+end handler
+----------
+
+----------
+-- this handler returns a path
+private handler fetchPath(in pObject as string, in pInt as integer) as Path
+	
+	if pObject is "border" then
+		return rounded rectangle path of rectangle [0.5, 0.5, (the trunc of my width)-0.5, (the trunc of my height)-0.5] with radius 2
+	end if
+	
+end handler
+----------
+
+----------
+-- this handler returns a paint
+private handler fetchPaint(in pObject as string) as Paint
+	
+	if pObject is "lines" then
+		return solid paint with color [51/255, 153/255, 1]
+		
+	else if pObject is "label" then
+		return solid paint with color [51/255, 153/255, 1]
+		
+	else if pObject is "selected label" then
+		return solid paint with color [1,1,1]
+	
+	end if
+
+end handler
+----------
+
+----------
+-- this handler returns a font
+private handler fetchFont() as Font
+	
+	if mSegmentDisplay is "icon" then
+		return font "fontawesome" at size kIconSize
+		
+	else
+		if mTheme is "ios-8" then
+			return font "Helvetica Neue" at size kTextSize
```

Comment:
We could let this inherit the system font for now. This would make it look correct on the three desktop platforms when used in the property inspector.